File: backend_net/ChronosDtnNetApi/Controllers/AuthController.cs
// Por que: Importa os namespaces do ASP.NET Core para estruturação de controllers de API, CORS e validações.
using Microsoft.AspNetCore.Mvc;
// Por que: Importa os namespaces do .NET para tarefas assíncronas.
using System.Threading.Tasks;
// Por que: Importa o namespace das DTOs de login e resposta.
using ChronosDtnNetApi.DTOs;
// Por que: Importa a interface do serviço de autenticação.
using ChronosDtnNetApi.Services;

// Por que: Define o namespace Controllers para organizar as portas de entrada HTTP da API.
namespace ChronosDtnNetApi.Controllers;

// Por que: Sinaliza que a classe responde chamadas de API, aplicando validação automática de ModelState.
[ApiController]
// Por que: Define o padrão de rotas "/api/auth" para este controlador.
[Route("api/auth")]
public class AuthController : ControllerBase
{
    // Por que: Mantém a referência privada ao serviço de autenticação de operadores.
    private readonly IAuthService _authService;

    // Por que: Construtor injetando o serviço de autenticação.
    public AuthController(IAuthService authService)
    {
        // Por que: Inicializa o serviço privado.
        _authService = authService;
    }

    // Por que: Endpoint POST /api/auth/login que valida credenciais e emite o token JWT para o operador.
    [HttpPost("login")]
    public ActionResult<TokenResponse> Login([FromBody] LoginRequest request)
    {
        // Por que: Invoca o serviço de autenticação para validar o operador.
        var response = _authService.Authenticate(request);
        // Por que: Retorna HTTP 200 OK com o token de autorização.
        return Ok(response);
    }
}

File: backend_java/src/main/java/br/com/fiap/chronosdtn/controller/AuthController.java
package br.com.fiap.chronosdtn.controller;

// Por que: Importa os DTOs de login, cadastro e resposta de token.
import br.com.fiap.chronosdtn.dto.LoginRequest;
import br.com.fiap.chronosdtn.dto.RegisterRequest;
import br.com.fiap.chronosdtn.dto.TokenResponse;
// Por que: Importa o serviço de autenticação de operadores.
import br.com.fiap.chronosdtn.service.AuthService;
// Por que: Importa anotações de validação e Web do Spring MVC.
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

// Por que: RestController sinaliza que a classe responde chamadas REST em formato JSON.
@RestController
// Por que: Mapeia o prefixo padrão das rotas de autenticação.
@RequestMapping("/api/auth")
// Por que: Habilita CORS para permitir que qualquer cliente (inclusive React Native) chame este endpoint.
@CrossOrigin(origins = "*", allowedHeaders = "*")
public class AuthController {

    // Por que: Dependência do serviço de login.
    private final AuthService authService;

    // Por que: Construtor injetado pelo Spring.
    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    // Por que: Mapeia requisições HTTP POST para efetuar o login do operador.
    @PostMapping("/login")
    public ResponseEntity<TokenResponse> login(@Valid @RequestBody LoginRequest request) {
        // Por que: Executa a autenticação e gera o token JWT correspondente.
        TokenResponse response = authService.authenticate(request);
        // Por que: Retorna status HTTP 200 OK contendo o token no corpo.
        return ResponseEntity.ok(response);
    }

    // Por que: Mapeia requisições HTTP POST para cadastrar um novo operador operacional.
    @PostMapping("/register")
    public ResponseEntity<String> register(@Valid @RequestBody RegisterRequest request) {
        // Por que: Executa a gravação do operador no banco relacional.
        authService.register(request);
        // Por que: Retorna status HTTP 201 Created com mensagem de sucesso.
        return new ResponseEntity<>("Operador cadastrado com sucesso!", HttpStatus.CREATED);
    }
}
